extends Control

var rng = RandomNumberGenerator.new()

const dedo_do_meio: String = " â€¦â€¦..â€¦../Â´Â¯/)â€¦â€¦â€¦.. (\\Â¯`\\\\ \n" \
+ "â€¦â€¦â€¦â€¦/â€¦.//â€¦â€¦â€¦.. â€¦\\\\â€¦.\\ \n" \
+ "â€¦â€¦â€¦../â€¦.//â€¦â€¦â€¦â€¦ â€¦.\\\\â€¦.\\ \n" \
+ "â€¦../Â´Â¯/â€¦./Â´Â¯\\â€¦â€¦â€¦../Â¯ `\\â€¦.\\Â¯`\\ \n" \
+ ".././â€¦/â€¦./â€¦./.|_â€¦â€¦_| .\\â€¦.\\â€¦.\\â€¦\\.\\.. \n" \
+ "(.(â€¦.(â€¦.(â€¦./.)..)..(..(. \\â€¦.)â€¦.)â€¦.).) \n" \
+ ".\\â€¦â€¦â€¦â€¦â€¦.\\/â€¦/â€¦.\\. ..\\/â€¦â€¦â€¦â€¦â€¦./ \n" \
+ "..\\â€¦â€¦â€¦â€¦â€¦.. /â€¦â€¦..\\â€¦â€¦â€¦â€¦â€¦..â€¦/ \n" \
+ "â€¦.\\â€¦â€¦â€¦â€¦..(â€¦â€¦â€¦. ..)â€¦â€¦â€¦â€¦â€¦./"

const palhaco: String = "ðŸ¤¡Ã…Å¤Ã…QÅ®Ä˜ ÄÃ˜Â§ PÃ…Ä½HÃ…Ã‡Ã˜Â§ Ä½Ã˜Ä¶Ã˜ðŸ¤¡    AGORA Ã‰ NOIS QUE MANDA NESSA PORRA \n" \
+ "â˜£â˜£â˜£ðŸ‘¿ \n" \
+ "ðŸ¤¡SAIAM DO GINÃSIO ðŸ¤¡  \n" \
+ " \n" \
+ "COMEÃ‡OU O ATAQUE ðŸ¤¡ðŸ¤¡ðŸ¤¡ \n" \
+ " \n" \
+ "HÃ! HÃ! HÃ! HÃ! HÃ!  \n" \
+ " \n" \
+ "ðŸ¤¡ðŸ¤¡ðŸ¤¡ðŸ¤¡ðŸ¤¡ðŸ¤¡ \n"

const shanub: String = "  **********************||\\ \n " \
+ "|SHANUB.ENCOMENDAS    |||**\\__ \n " \
+ "|_____________________|||___|_) \n " \
+ "!(@)'(@)*******(@)'(@)*****(@)[ \n "

const cinco_vezes: String = "VocÃª jÃ¡ se fudeu 5 vezes nesse ginÃ¡sio. Agora vou me ausentar pois tenho reuniÃ£o. Infelizmente sou um homem muito dedicado no trabalho e isso me faz ser uma pessoa solicitada e ocupada"
const dez_vezes: String = "Voltando aqui pra falar: VocÃª jÃ¡ se fudeu 10 vezes nesse ginÃ¡sio"
const quinze_vezes: String = "Meu Deus vocÃª jÃ¡ se fudeu [font_size=50]15[/font_size] vezes nesse ginÃ¡sio. InacreditÃ¡vel"
const recorde_vezes: String = "Opa, voltei aqui sÃ³ pra avisar: vocÃª jÃ¡ se fudeu [x] vezes nesse ginÃ¡sio. Ã‰ um recorde! ParabÃ©ns pela Ãºnica conquista que vocÃª vai receber na sua vida"
const recordissimo_vezes: String = "VocÃª jÃ¡ se fudeu [x] vezes nesse ginÃ¡sio. E vocÃª continua destruindo absolutamente o recorde. Quem diria que vocÃª ia se tornar especial em alguma coisa"

const MENSAGENS: Array = [
	"VocÃª Ã© fraco e nÃ£o vai resistir ao inverno",
	"saidfhnadojgiasdgnIUSAHGFIASDGS9ODGFHJSDA9UGASJGKOASÂ´DGAS",
	"KKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKK",
	dedo_do_meio,
	"Mas bah tu Ã© um baita dum chinelÃ£o mesmo",
	"VocÃª devia desistir. No geral",
	"Liguei para sua mÃ£e e ela disse que preferia nÃ£o ter criado um filho. E que vocÃª Ã© adotado",
	"Viu, sÃ³ te avisar que psicografamos uma carta do seu pai. Ele falou que o inferno Ã© menos pior do que parece",
	"VocÃª Ã© menor que a bola e ela ainda consegue te acertar",
	"Ah sei lÃ¡ pensa vocÃª agora em uma ofensa aÃ­. do jeito que vocÃª se odeia vocÃª vai pensar em uma rapidinho",
	"TÃ¡ de corpo mole Ã©?",
	 "Levanta essa bunda do chÃ£o e tenta de novo",
	 "VocÃª gosta da JÃºlia, nÃ©? O Lauro tÃ¡ ficando com ela. E ela falou que vocÃª Ã© estranho",
	"JÃ¡ te falaram que vocÃª anda rebolando? VocÃª devia andar mais reto e de forma menos estranha",
	"Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
	"sÃ³ regresso irmÃ£o, muitos vao te derrubar mas, aconteÃ§a oq acontecer, nÃ£o levante. te admiro pois vc comeÃ§ou do zero e continuou la, te criticar Ã© facil, difÃ­cil Ã© elogiar. tudo deles nada nosso ðŸ‘ŠðŸ»ðŸ”¥",
	"Eu vou deixar registrado aqui pra vocÃª, seu bosta. Seu obtuso. Seu ser maldito. Que tem gente aqui nesse grupo que tÃ¡ hÃ¡ anos na UFRGS que tÃ¡ se fudendo, entendeu. Que quer se formar de uma vez, entendeu. Isso aqui Ã© SemÃ¢ntica Formal. Ã‰ uma cadeira do quinto semestre, que Ã© dum semestre que tem mais cadeiras obrigatÃ³rias nessa merda de curso, no caso, a ciÃªncia da computaÃ§Ã£o. Imagino entÃ£o o que que o pessoal da engenharia passa, tÃ¡. Mas sÃ³ vou te dizer o seguinte: a gurizada aqui se fudendo no EAD aqui Ã© uma bosta, sabe, Ã© uma merda. As pessoas querendo tocar a vida, se formar, conseguir um emprego? Tentando dar um jeito de conciliar os estudos. E vocÃª. Seu bosta. Seu lixo. Em vez de nÃ£o ajudar... nÃ£o quer ajudar, cara. Fica na tua. Entendeu? Faz os teus trabalhos sozinho, se tu Ã© fodÃ£o. Se tu Ã© pica das galÃ¡xias. Se tu vai ser laureado, seu merda.",
	"Curse of RA ð“€€ ð“€ ð“€‚ ð“€ƒ ð“€„ ð“€… ð“€† ð“€‡ ð“€ˆ ð“€‰ ð“€Š ð“€‹ ð“€Œ ð“€ ð“€Ž ð“€ ð“€ ð“€‘ ð“€’ ð“€“ ð“€” ð“€• ð“€– ð“€— ð“€˜ ð“€™ ð“€š ð“€› ð“€œ ð“€ ð“€ž ð“€Ÿ ð“€  ð“€¡ ð“€¢ ð“€£ ð“€¤ ð“€¥ ð“€¦ ð“€§ ð“€¨ ð“€© ð“€ª ð“€« ð“€¬ ð“€­ ð“€® ð“€¯ ð“€° ð“€± ð“€² ð“€³ ð“€´ ð“€µ ð“€¶ ð“€· ð“€¸ ð“€¹ ð“€º ð“€» ð“€¼ ð“€½ ð“€¾ ð“€¿ ð“€ ð“ ð“‚ ð“ƒ ð“„ ð“… ð“† ð“‡ ð“ˆ ð“‰ ð“Š ð“‹ ð“Œ ð“ ð“Ž ð“ ð“ ð“‘ ð“€„ ð“€…",
	"Escalador de calÃ§ada",
	"Arquiteto de Lego",
	"Escultor de suspiro",
	"Goleiro de pebolim",
	"Tampinha",
	"Operador de zÃ­per",
	"Mordedor de canela",
	"VocÃª Ã© o menor dos meus problemas",
	"MÃ£o de alface",
	"Raquete furada",
	palhaco,
	shanub,
	"Deus estÃ¡ insatisfeito com a sua existÃªncia e vocÃª deve se preparar para uma eternidade em sofrimento",
	"Lolzeiro",
	"Viu, Jones me contou uma coisa depois da aula de quÃ­mica de vocÃªs. SeriÃ£o que vocÃª ainda Ã© BV?",
	"ðŸš¨ðŸš¨ URGENTE!!!ðŸš¨ðŸš¨ ðŸš¨ðŸš¨FOI DETECTADO O PRIMEIRO CASO DA DOENÃ‡A DO PASSARO ðŸ¦…ðŸ¦…ðŸ¦… NA ESCOLA CAIXEIRO VIAJANTE ðŸš¨ðŸš¨ PASSARO O PAU NO ALUNO PEDRO",
	"Oi meu nome Ã© Samara, tenho 14 anos (Teria se estivesse viva), morri aos 13 em Cascavel-PR. Eu andava de bicicleta Quando nÃ£o pude desviar de um arame farpado. O pior foi que o dono do lote nÃ£o quis me ajudar, riu bastante mim apÃ³s agonizar por 2 horas enroscada no arame eu faleci, atravÃ©s dessa mensagem eu peÃ§o que faÃ§am com que eu possa descanÃ§ar em paz. Envie isso para 20 comunidades e minha alma estara sendo salva por vocÃª e pelos outros 20 que receberÃ£o. Caso nÃ£o repasse essa mensagem vou visitar-lhe hoje a noite assim vc poderÃ¡ conhecer o tal arame bem de pertinho. Dia 15 de Julho Mariana resolveu rir dessa mensagem, uma noite depois ela sumiu sem deixar vestigios. O mesmo aconteceu com Kare dia 18 de Outubro. NÃ£o Quebre essa corrente por favor, a nÃ£o ser que queira sentir a minha presenÃ§a",
	"O Naruto pode ser um pouco duro Ã s vezes, talvez vocÃª nÃ£o saiba disso, mas o Naruto tambÃ©m cresceu sem pai. Na verdade ele nunca conheceu nenhum de seus pais, e nunca teve nenhum amigo em nossa aldeia. Mesmo assim eu nunca vi ele chorar, ficar zangado ou se dar por vencido, ele estÃ¡ sempre disposto a melhorar, ele quer ser respeitado, Ã© o sonho dele e o Naruto daria a vida por isso sem hesitar. Meu palpite Ã© que ele se cansou de chorar e decidiu fazer alguma coisa a respeito!",
	"Eu entendo, fica difÃ­cil mesmo ver a bola quando vocÃª tem que olhar pra cima pra enxergar. Pintorzinho de rodapÃ©",
	"CabeÃ§a de bolota cabeÃ§a de barril cabeÃ§a de abÃ³bora cabeÃ§a de fuzil Ã© o cabeÃ§Ã£o de plÃ¡stico cabeÃ§a de Max Steel",
	"Cu baixo",
	"Tamburete",
	"AnÃ£o de jardim",
	"Dunga",
	"Toco de amarrar bode",
	"BebÃª reborn",
	"Hipista de cachorro",
	"CabeÃ§a de desequilÃ­brio",
	"FALAAAAAAAAAAAAAAAAAAAA DERICK AHAHAHAHAHAHAHAA",
	"AHAHAHAHAHAHAHAAAHAHAHAHAHAHAHAAAHAHAHAHAHAHAHAAAHAHAHAHAHAHAHAAAHAHAHAHAHAHAHAAAHAHAHAHAHAHAHAAAHAHAHAHAHAHAHAAAHAHAHAHAHAHAHAAAHAHAHAHAHAHAHAAAHAHAHAHAHAHAHAAAHAHAHAHAHAHAHAAAHAHAHAHAHAHAHAAAHAHAHAHAHAHAHAAAHAHAHAHAHAHAHAAAHAHAHAHAHAHAHAAAHAHAHAHAHAHAHAAAHAHAHAHAHAHAHAAAHAHAHAHAHAHAHAA"
]

const MENSAGENS_CHUMBO: Array = [
	"Caiu no conto da bola de chumbo nÃ© otÃ¡rio",
	"Pqp caiu de novo"
]

const MENSAGENS_BOLICHE: Array = [
	"STRIKE",
	"Bom que a bola de boliche ainda Ã© mais macia que esse cabeÃ§Ã£o de bigorna"
]

const MENSAGENS_TINTA: Array = [
	"fdsifuhsrBGDASUIGHADSGA ESCORREGOU QUE NEM UM MERDA",
	"Pqp caiu de novo",
	"Ã‰ difÃ­cil mesmo manter o equilÃ­brio com essa cabeÃ§a de playmobil"
]

const SPEED = 12

var chosen_text : String = ""
var visible_characters = 0

# Called when the node enters the scene tree for the first time.
func _ready():
	var index = rng.randf_range(0, MENSAGENS.size() - 1)
	chosen_text = MENSAGENS[index]
	pass # Replace with function body.


# Called every frame. 'delta' is the elapsed time since the previous frame.
func _process(delta):
	visible_characters += ceil(SPEED * delta)
	if (visible_characters > chosen_text.length()):
		visible_characters = chosen_text.length()
	$CanvasLayer/RichTextLabel.text = "[font_size=25]" + chosen_text.substr(0, visible_characters) + "[/font_size]"
	
	if(Input.is_action_just_pressed("Enter") or Input.is_action_just_pressed("Esc")):
		Global.game_controller.reload_current_2d_scene()
		
	if Input.is_action_just_pressed("Restart"):
		Global.game_controller.reload_current_2d_scene()
